### ISSUE TITLE
Wrong number or type of arguments for overloaded function 'Matrix_sparseCopyFrom'

### ISSUE THREAD

**Issue by iyishengyoun** [PaddlePaddle/Paddle#2911]
![image](https://user-images.githubusercontent.com/10525315/28257940-c89255f0-6b00-11e7-9ff0-8ccc1d4c4a0b.png)
输入数据：yield [cuid_vec_dict.values(), nid_vec_dict.values()], [label]
cuid_vec_dict.values()和nid_vec_dict.values()是[0,0,1.......]0和1构成的长度为90000多的list
label是0或1

paddle配置信息：
```python
x = paddle.layer.data(name='x', type=paddle.data_type.sparse_binary_vector(2))
y_predict = paddle.layer.fc(input=x, size=1, act=paddle.activation.Linear())
y = paddle.layer.data(name='y', type=paddle.data_type.sparse_binary_vector(1))
cost = paddle.layer.mse_cost(input=y_predict, label=y)
parameters = paddle.parameters.create(cost)
optimizer = paddle.optimizer.Momentum(momentum=0, learning_rate=0.01)
trainer = paddle.trainer.SGD(cost=cost, parameters=parameters, update_equation=optimizer)
```

**Comment by lcy-seso** [comment]
您好，请您谅解，尽量不要在issue 中贴图。您可以把所有的出错信息直接复制到issue中。

**Comment by Yancey1989** [comment]
@iyishengyoun 麻烦贴全一些吧，比如`feeder`相关的代码也贴一下？

**Comment by lcy-seso** [comment]
嗯，这个错误是因为 reader 返回的数据有误，请提供 `reader`的信息，否则无法判断。
有一个疑点，按照上面的描述，定义x时的size ，不应该是2，而是这个稀疏输入的维度。
```python
x = paddle.layer.data(name='x', type=paddle.data_type.sparse_binary_vector(2))
```

**Comment by iyishengyoun** [comment]
@Yancey1989 @lcy-seso  多谢，已经排查出问题了，data_type用错了，知晓了再遇到问题不贴图了

**Comment by christianahui** [comment]
请问data_type哪里错了？我遇到了相同的问题，针对0101这种稀疏的数据类型不应该用sparse_binary_vector吗？ @iyishengyoun
